systems/controllers/osc/operational_space_control.h

```diff
@@ -0,0 +1,246 @@
+///     drake::trajectories::Trajectory<double>
+/// such as `PiecewisePolynomial` and `ExponentialPlusPiecewisePolynomial`.
+/// The users can connect the output ports of the desired trajecotry blocks to
+/// the corresponding input ports of `OperationalSpaceControl` by using
+/// the method get_tracking_data_input_port().
+
+/// The procedure of setting up `OperationalSpaceControl`:
+///   1. create an instance of `OperationalSpaceControl`
+///   2. add costs/constraints/desired trajectories
+///   3. call BuildOSC()
+///   4. (if the users created desired trajectory blocks by themselves) connect
+///      `OperationalSpaceControl`'s input ports to corresponding output ports
+///      of the trajectory source.
+
+class OperationalSpaceControl : public drake::systems::LeafSystem<double> {
+ public:
+  OperationalSpaceControl(const RigidBodyTree<double>& tree_w_spr,
+                          const RigidBodyTree<double>& tree_wo_spr,
+                          bool used_with_finite_state_machine = true,
+                          bool print_tracking_info = false);
+
+  // Input/output ports
+  const drake::systems::InputPort<double>& get_robot_output_input_port() const {
+    return this->get_input_port(state_port_);
+  }
+  const drake::systems::InputPort<double>& get_fsm_input_port() const {
+    return this->get_input_port(fsm_port_);
+  }
+  const drake::systems::InputPort<double>& get_tracking_data_input_port(
+      std::string name) const {
+    return this->get_input_port(traj_name_to_port_index_map_.at(name));
+  }
+
+  // Cost methods
+  void SetInputCost(Eigen::MatrixXd W) {W_input_ = W;}
+  void SetAccelerationCostForAllJoints(Eigen::MatrixXd W) {W_joint_accel_ = W;}
+  void AddAccelerationCost(std::string joint_vel_name, double w);
+
+  // Constraint methods
+  void DisableAcutationConstraint() {with_input_constraints_ = false;}
+  void SetContactFriction(double mu) {mu_ = mu;}
+  void SetWeightOfSoftContactConstraint(double w_soft_constraint) {
+    w_soft_constraint_ = w_soft_constraint;
+  }
+  void AddContactPoint(std::string body_name, Eigen::VectorXd pt_on_body);
+  void AddStateAndContactPoint(int state,
+                               std::string body_name,
+                               Eigen::VectorXd pt_on_body);
+
+  // Tracking data methods
+  /// The third argument is used to set a period in which OSC does not track the
+  /// desired traj (the period starts when the finite state machine switches to
+  /// a new state)
+  void AddTrackingData(OscTrackingData* tracking_data, double duration = 0);
+  void AddConstTrackingData(OscTrackingData* tracking_data, Eigen::VectorXd v,
+      double duration = 0);
+  std::vector<OscTrackingData*>* GetAllTrackingData(){
+    return tracking_data_vec_.get();
+  }
+  OscTrackingData* GetTrackingDataByIndex(int index) {
+    return tracking_data_vec_->at(index);
+  }
+
+  // Osc leafsystem builder
+  void BuildOSC();
```

Comment:
Why not just `Build`? The class is `OSC`, so it's unnecessary to specify that twice.

Reply:
Right, Build sounds better